pkg/controller/registry/reconciler/reconciler.go

```diff
@@ -195,19 +192,29 @@ func Pod(source *operatorsv1alpha1.CatalogSource, name string, image string, saN
 			NodeSelector: map[string]string{
 				"kubernetes.io/os": "linux",
 			},
-			SecurityContext: &corev1.PodSecurityContext{
-				SeccompProfile: &corev1.SeccompProfile{
-					Type: corev1.SeccompProfileTypeRuntimeDefault,
-				},
-			},
 			ServiceAccountName: saName,
 		},
 	}
 
-	if runAsUser > 0 {
-		pod.Spec.SecurityContext.RunAsUser = &runAsUser
-		pod.Spec.SecurityContext.RunAsNonRoot = pointer.Bool(true)
+	if source.Spec.GrpcPodConfig != nil && source.Spec.GrpcPodConfig.RequirePrivilegedAccess {
+		pod.Spec.Containers[0].SecurityContext.AllowPrivilegeEscalation = pointer.Bool(true)
+	} else {
+		log.Printf("(anik120) setting everything")
+		pod.Spec.Containers[0].SecurityContext.AllowPrivilegeEscalation = pointer.Bool(false)
+		pod.Spec.Containers[0].SecurityContext.Capabilities = &corev1.Capabilities{
+			Drop: []corev1.Capability{"ALL"},
+		}
+		pod.Spec.SecurityContext = &corev1.PodSecurityContext{
+			SeccompProfile: &corev1.SeccompProfile{
+				Type: corev1.SeccompProfileTypeRuntimeDefault,
+			},
+		}
+		if runAsUser > 0 {
```

Comment:
when running legacy catalogs you're not going to respect this value?  (who sets this? where does it get set? why would they set it?)

Reply:
The value for this is dependent on an [argument to the container](https://github.com/openshift/operator-framework-olm/pull/349/files#diff-3095edb22e78d950b5bb2d52942b13bad32f4b097451d907aa04cb69ad658985R66), that is determined [here](https://github.com/openshift/operator-framework-olm/pull/349/files#diff-3a59260edccecabbc7c818c855c7b35842572551f5039cf4e6590f939beb81bfR87-R90). We decided to introduce that argument so that we can set that to `true` for upstream deployment, and `false` for downstream deployment, and then set a default userID for when that flag's value is `true`, because we wanted to keep control of what userID the container's run with (in absence of a requirement for a configurable userID). 


For legacy catalogs, [we never set these fields](https://github.com/openshift/operator-framework-olm/blob/release-4.10/staging/operator-lifecycle-manager/pkg/controller/registry/reconciler/reconciler.go#L116-L180).  So even if the flag `set-workload-user-id` is set to `true`, when `catsrc.spec.grpcPodConfig.requirePrivilegedAccess` is set to true, that is an indication that we don't need to set the userID to the default value (instead, let whatever used to happen in 4.10, happen. Next comment has logs for an example)

Reply:
```
$ k get ns test-ns -o yaml | grep labels: -A 3
  labels:
    kubernetes.io/metadata.name: test-ns
    pod-security.kubernetes.io/enforce: privileged
    pod-security.kubernetes.io/enforce-version: v1.24

$ k get pods -n test-ns 
NAME                READY   STATUS    RESTARTS   AGE
ditto-index-4cnmf   1/1     Running   0          23h

$ k get pods -n test-ns -o yaml | grep securityContext -A 2
      securityContext:
        allowPrivilegeEscalation: true
        readOnlyRootFilesystem: false
--
    securityContext: {}
    serviceAccount: ditto-index
    serviceAccountName: ditto-index

$ k get pods -n test-ns -o yaml | grep containerStatuses -A 11
    containerStatuses:
    - containerID: containerd://00e416a3797d00b2b29a4059cec21511134b1e113afd0acfe629c75ac5c0cd19
      image: docker.io/anik120/ditto-index-opm:v1.21.0
      imageID: docker.io/anik120/ditto-index-opm@sha256:36a8daca4365b24bb8699e5f89678dad36a7b4a01127963bf289afd67ecde96a
      lastState: {}
      name: registry-server
      ready: true
      restartCount: 0
      started: true
      state:
        running:
          startedAt: "2022-09-06T19:30:32Z"

```

Reply:
got it